model/criteria/operators.go

```diff
@@ -227,3 +228,46 @@ func inPeriod(m map[string]interface{}, negate bool) (Expression, error) {
 func startOfPeriod(numDays int64, from time.Time) string {
 	return from.Add(time.Duration(-24*numDays) * time.Hour).Format("2006-01-02")
 }
+
+type InPlaylist map[string]interface{}
+
+func (ipl InPlaylist) ToSql() (sql string, args []interface{}, err error) {
+	return inList(ipl, false)
+}
+
+func (ipl InPlaylist) MarshalJSON() ([]byte, error) {
+	return marshalExpression("inPlaylist", ipl)
+}
+
+type NotInPlaylist map[string]interface{}
+
+func (ipl NotInPlaylist) ToSql() (sql string, args []interface{}, err error) {
+	return inList(ipl, true)
+}
+
+func (ipl NotInPlaylist) MarshalJSON() ([]byte, error) {
+	return marshalExpression("notInPlaylist", ipl)
+}
+
+func inList(m map[string]interface{}, negate bool) (sql string, args []interface{}, err error) {
+	var playlistname string
+	var ok bool
+	if playlistname, ok = m["name"].(string); !ok {
+		return "", nil, errors.New("playlist name not given")
+	}
+
+	// Subquery to check if media_file.id matches with the playlist name in playlist_tracks table
+	subQuery := squirrel.Select("1").
+		From("playlist_tracks pl").
+		LeftJoin("playlist on pl.playlist_id = playlist.id").
+		Where(squirrel.And{squirrel.Expr("pl.media_file_id = media_file.id"), squirrel.Eq{"playlist.name": playlistname}})
```

Comment:
An alternative would be
```sql
media_file.id IN (
  SELECT media_file_id
  FROM playlist_tracks
  LEFT JOIN playlist
  ON playlist_tracks.playlist_id = playlist.id
  WHERE playlist.name = "<name>"
)
```

I'm not an SQL expert so I can't tell definitively whether it would be better, but my gut feeling is that it should be easier to optimize for the database.

Reply:
Adding on here: This is **significantly** faster. Here's what I used to replicate your results
```go
// Subquery to check if media_file.id matches with the playlist name in playlist_tracks table
subQuery := squirrel.Select("media_file_id").
	From("playlist_tracks pl").
	LeftJoin("playlist on pl.playlist_id = playlist.id").
	Where(squirrel.Eq{"playlist.name": playlistname})
subQText, subQArgs, err := subQuery.PlaceholderFormat(squirrel.Question).ToSql()
if err != nil {
	return "", nil, err
}
if negate {
	return "media_file.id IN (" + subQText + ")", subQArgs, nil
} else {
	return "media_file.id IN (" + subQText + ")", subQArgs, nil
}
```

Performance:
OLD: 947.9 ms
```
PM GO.1 |  TRAC[0216] SQL: `INSERT INTO playlist_tracks (id,playlist_id,media_file_id) SELECT row_number() over (order by random()) as id, '...' as playlist_id, media_file.id as media_file_id FROM media_file LEFT JOIN annotation on (annotation.item_id = media_file.id AND annotation.item_type = 'media_file' AND annotation.user_id = '...') LEFT JOIN media_file_genres ag on media_file.id = ag.media_file_id LEFT JOIN genre on ag.genre_i
8:58:23 PM GO.1 |  >  d = genre.id WHERE (EXISTS(SELECT 1 FROM playlist_tracks pl LEFT JOIN playlist on pl.playlist_id = playlist.id WHERE (pl.media_file_id = media_file.id AND playlist.name = ?))) GROUP BY media_file.id ORDER BY random()`  args="[[...],<nil>]" elapsedTime=947.9ms requestId=ok/0AOnerGlUd-000103 rowsAffected=4282
```
NEW: 34 ms
```
PM GO.1 |  TRAC[0004] SQL: `INSERT INTO playlist_tracks (id,playlist_id,media_file_id) SELECT row_number() over (order by random()) as id, '...' as playlist_id, media_file.id as media_file_id FROM media_file LEFT JOIN annotation on (annotation.item_id = media_file.id AND annotation.item_type = 'media_file' AND annotation.user_id = '...') LEFT JOIN media_file_genres ag on media_file.id = ag.media_file_id LEFT JOIN genre on ag.genre_i
8:59:14 PM GO.1 |  >  d = genre.id WHERE (media_file.id IN (SELECT media_file_id FROM playlist_tracks pl LEFT JOIN playlist on pl.playlist_id = playlist.id WHERE playlist.name = ?)) GROUP BY media_file.id ORDER BY random()`  args="[[...],<nil>]" elapsedTime=33.4ms  rowsAffected=4282
```

Reply:
Playlist JSON used for testing:
```json
{
    "all":  [
        {"inPlaylist": {"name": "..." }}
    ],  
    "sort": "random"
}
```